### PR TITLE
Upgrade more methods and fix view issue

### PR DESCRIPTION
**Pull request by pentacular**
viewId was incorrectly receiving options in some cases

### CODE REVIEW

**Review by BarbourSmith — APPROVED** [review]
👍 👍 Looks good to me!

File: algorithm/cgal/fromPolygonSoup.js [hunk 1]
```diff
@@ -24,6 +24,13 @@ export const fromPolygonSoup = (
       cornerThreshold,
     })}`
   );
+  if (cornerThreshold === 0) {
+    try {
```
Comment (BarbourSmith) [inline algorithm/cgal/fromPolygonSoup.js:28]: What does this do?

File: algorithm/cgal/FromPolygonSoup.h [hunk 2]
```diff
@@ -22,32 +28,32 @@ int FromPolygonSoup(Geometry* geometry, emscripten::val fill) {
...
-        std::vector<Face_index> faces;
-        CGAL::Polygon_mesh_processing::triangulate_hole(
-            mesh, edge, std::back_inserter(faces),
-            CGAL::parameters::use_2d_constrained_delaunay_triangulation(false));
-        if (faces.empty()) {
-          failed = true;
-        }
-        break;
-      }
+  if (wrap_always || CGAL::Polygon_mesh_processing::does_self_intersect(mesh)) {
```
Comment (BarbourSmith) [inline algorithm/cgal/FromPolygonSoup.h:31]: How fast is wrapping?

File: algorithm/cgal/computeOrientedBoundingBox.js [hunk 3]
```diff
@@ -6,25 +6,29 @@ export const computeOrientedBoundingBox = (inputs) => {
...
-        );
-      case STATUS_EMPTY:
-        return;
-      default:
-        throw new Error(`Unexpected status ${status}`);
+  return withCgalGeometry(
+    'computeOrientedBoundingBox',
+    inputs,
+    (cgalGeometry, g) => {
+      const status = g.ComputeOrientedBoundingBox(cgalGeometry);
```
Comment (BarbourSmith) [inline algorithm/cgal/computeOrientedBoundingBox.js:13]: My brain read this as "Computer oriented bounding box" the first time through and I was like "What is the computer frame of reference? Doesn't it move with the camera? 🙃

File: api/shape/shadow.js [hunk 4]
```diff
@@ -4,9 +4,15 @@ import { cast as castGeometry } from '@jsxcad/geometry';
 
 export const shadow = Shape.registerMethod2(
   'shadow',
```
Comment (BarbourSmith) [inline api/shape/shadow.js:6]: Did this used to be called project? I feel like the name shadow might imply some sort of parallax where there is a point light source...although maybe I'm over thinking it and 'project' has the same issue

**Review by BarbourSmith — APPROVED** [review]
Looks great to me!

File: api/shape/shadow.js (hunk 4 above)
Comment (pentacular) [reply api/shape/shadow.js:6]: I think it probably ought to be silhouette(). wdyt?

File: algorithm/cgal/FromPolygonSoup.h (hunk 2 above)
Comment (pentacular) [reply algorithm/cgal/FromPolygonSoup.h:31]: On my machine, it's tens of seconds for most shapes, which isn't fast, but is usable.

File: algorithm/cgal/fromPolygonSoup.js (hunk 1 above)
Comment (pentacular) [reply algorithm/cgal/fromPolygonSoup.js]: Skips simplification if all corners should be preserved.

File: api/shape/shadow.js (hunk 4 above)
Comment (BarbourSmith) [reply api/shape/shadow.js:6]: Brilliant. That's exactly the right word

File: algorithm/cgal/fromPolygonSoup.js (hunk 1 above)
Comment (BarbourSmith) [reply algorithm/cgal/fromPolygonSoup.js]: but why does it throw an error and then catch it?

Comment (pentacular) [reply algorithm/cgal/fromPolygonSoup.js]: Ah, it gives you a stack trace to figure out how you got there. :)

I was trying to figure out what wasn't propagating the threshold through properly.

**Pull request merged**